src/MICore/ProcessMonitor.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+
+namespace MICore
+{
+    public class ProcessMonitor: IDisposable
+    {
+        private readonly TimeSpan EXIT_POLL_DELTA = TimeSpan.FromMilliseconds(200);
+        private int _processId;
+        private Timer _exitMonitorTimer;
+
+        public ProcessMonitor(int processId)
+        {
+            if (!PlatformUtilities.IsLinux() && !PlatformUtilities.IsOSX())
+            {
+                throw new NotImplementedException();
+            }
+
+            _processId = processId;
+        }
+
+        public void Start()
+        {
+            _exitMonitorTimer = new Timer(MonitorForExit, null, TimeSpan.FromMilliseconds(0), EXIT_POLL_DELTA);
+        }
+
+        public event EventHandler ProcessExited;
+
+        private bool HasExited()
+        {
+            return !UnixUtilities.IsProcessRunning(_processId);
+        }
+
+        private void MonitorForExit(object o)
+        {
+            if (HasExited())
+            {
+                _exitMonitorTimer.Change(Timeout.Infinite, Timeout.Infinite);
```

Comment:
You are right. I had the idea in my mind that the same `ProcessMonitor` could be reused even after the process has exited. I hadn't taken into account the fact that `_processId` can't be changed and there's no point in doing so.